addons/survey/static/src/js/survey_result.js

```diff
@@ -1,42 +1,102 @@
+var ajax = require('web.ajax');
+var publicWidget = require('web.public.widget');
+
+// The given colors are the same as those used by D3
+var D3_COLORS = ["#1f77b4","#ff7f0e","#aec7e8","#ffbb78","#2ca02c","#98df8a","#d62728",
+                    "#ff9896","#9467bd","#c5b0d5","#8c564b","#c49c94","#e377c2","#f7b6d2",
+                    "#7f7f7f","#c7c7c7","#bcbd22","#dbdb8d","#17becf","#9edae5"];
+
+publicWidget.registry.SurveyResultWidget = publicWidget.Widget.extend({
+    selector: '.o_survey_result',
+    events: {
+        'click td.survey_answer i.fa-filter': '_onSurveyAnswerFilterClick',
+        'click .clear_survey_filter': '_onClearFilterClick',
+        'click span.filter-all': '_onFilterAllClick',
+        'click span.filter-finished': '_onFilterFinishedClick',
+    },
+
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
+
+    /**
+    * @override
+    */
+    willStart: function () {
+        var url = '/web/webclient/locale/' + (document.documentElement.getAttribute('lang') || 'en_US').replace('-', '_');
+        var localeReady = ajax.loadJS(url);
+        return Promise.all([this._super.apply(this, arguments), localeReady]);
+    },
+
+    /**
+    * @override
+    */
+    start: function () {
+        var self = this;
+
+        //Script For Pagination
+        var survey_pagination = $('.pagination');
+        $.each(survey_pagination, function(index, pagination){
+            var question_id = $(pagination).attr("data-question_id");
+            var limit = $(pagination).attr("data-record_limit"); //Number of Record Par Page. If you want to change number of record per page, change record_limit in pagination template.
+            self.$('#table_question_'+ question_id +' tbody tr:lt('+limit+')').removeClass('d-none');
+            self.$('#pagination_'+question_id+' li a').click(function(event){
+                event.preventDefault();
+                self.$('#pagination_'+question_id+' li').removeClass('active');
+                $(this).parent('li').addClass('active');
+                self.$('#table_question_'+ question_id +' tbody tr').addClass('d-none');
+                var num = $(this).text();
+                var min = (limit * (num-1))-1;
+                if (min == -1){
+                    self.$('#table_question_'+ question_id +' tbody tr:lt('+ limit * num +')').removeClass('d-none');
+                }
+                else{
+                    self.$('#table_question_'+question_id+' tbody tr:lt('+ limit * num +'):gt('+min+')').removeClass('d-none');
+                }
+            });
+            self.$('#pagination_'+question_id+' li:first').addClass('active').find('a').click();
+        });
+
+        //Script For Graph
+        var survey_graphs = self.$('.survey_graph');
+        $.each(survey_graphs, function(index, graph){
```

Comment:
Same here.
Could be a small `SurveyResultChartWidget` embedded in this file with nice and clean init/start methods and ...

Reply:
done in next commit (by awa btw)